src/node.cpp

```diff
@@ -1126,7 +1126,7 @@ void Node::setpubliclink(handle ph, m_time_t cts, m_time_t ets, bool takendown,
     if (!plink) // creation
     {
         assert(client->mPublicLinks.find(nodehandle) == client->mPublicLinks.end());
-        plink = new PublicLink(ph, cts, ets, takendown, authKey);
+        plink = new PublicLink(ph, cts, ets, takendown, authKey.c_str());
```

Comment:
`authKey` might be empty in some calls. Probaby we want to:
```suggestion
        plink = new PublicLink(ph, cts, ets, takendown, authKey.empty() ? nullptr : authKey.c_str());
```